src/pip/_internal/req/constructors.py

```diff
@@ -45,15 +46,6 @@
 operators = Specifier._operators.keys()
 
 
-def is_archive_file(name):
```

Comment:
pretty sure you would want to use existing impl instead of replacing it

Reply:
I'll create a new PR in which I will replace the implementation of the `is_archive_file` (as I believe it should be in the `utils` files and can be simplified).